### DISCUSSION TITLE
Tiled automapping-like at game runtime

### DISCUSSION THREAD

**Discussion by wiredmatt** [libktx/ktx#469] [General]
Hey! Looking into getting some randomness to my top-down levels. I know the base layer will always be the same, but it'd be nice to see grass, flowers, and other decorative assets placed randomly each time the maps are loaded. I'm looking to reproduce [this](https://doc.mapeditor.org/en/stable/manual/automapping/) from the Tiled editor itself, what do you guys think? I'm thinking about using their already defined rules format, and edit the layer for decorative assets of the map at runtime... The [API definition](https://github.com/libktx/ktx/tree/master/tiled#general) seems to allow one to do so, weekend is over for me but I'll come back as soon as I get some free time.

In the meantime, if anyone has any comments or suggestions (maybe this is already implemented) I'd highly appreciate it, have a nice week

**Comment by czyzby** [comment]
Seems like it should be possible at first glance, but please do share if you encounter any issues or if you think any utilities are missing in the KTX module.

Since this is not specific to KTX, you might want to visit the libGDX Discord, particularly the `#proc-gen`channel. I'm sure people will be happy to share ideas there. Cheers.

**Reply by wiredmatt** [reply]
Cheers! I'll check it out, I'm more concerned with changing the tiles at runtime rather than the algo, though. We'll see how it goes, I'll write back next weekend :) !
